### DISCUSSION TITLE
Add callback to watch for notes ending in a schedule?

### DISCUSSION THREAD

**Discussion by drscottlobo** [danigb/smplr#14] [Ideas]
Hi Danigb, thanks for your work on this! I'm excited to try it out! 

Would it be possible to add some sort of listener for the end of a note in a schedule? Maybe a callback onEnd? I'm writing some simple score playback and I want to color a note in my score while it's playing, then have it return to black once it's completed. Currently just running a separate clock (WAAclock) and running a separate schedule there to track the note durations, at the same time as the soundfont plays its schedule, so not ideal.

Would something like this be possible? 

`
const now = context.currentTime;

["C4", "E4", "G4", "C5"].forEach((note, i) => {
  piano.start({ note, time: now + i, duration: 0.5, onEnd: callbackFunction });
});
`

Thanks!

**Comment by danigb** [comment]
Yes, sure! I'll add that soon

**Comment by drscottlobo** [comment]
Awesome!! Thank you!_______________Scott Wolf, DMAOxnard College - Professor of MusicOn May 17, 2023, at 12:35 AM, danigb ***@***.***> wrote:﻿
Yes, sure! I'll add that soon

**Comment by danigb** [comment]
Implemented. Read here: https://github.com/danigb/smplr#onended-event

**Reply by drscottlobo** [reply]
Wow that was fast! It works perfectly! Thank you! 

Would it be too much to ask for an onStart callback as well? 

Also, how can I donate? It would be great to be able to buy you a _cafe con leche_?
